src/modules/control_allocator/ControlAllocator.cpp

```diff
@@ -398,16 +398,16 @@ ControlAllocator::Run()
 
 			_control_allocation[i]->clipActuatorSetpoint();
 		}
+	}
 
-		// Publish actuator setpoint and allocator status
-		publish_actuator_controls();
+	// Publish actuator setpoint and allocator status
```

Comment:
Does this really make a difference? `Run` is only called when either the torque or thrust setpoint topics update in the first place.

We can add some hardening (such as running on param update or a [backup scheduling](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/mixer_module/mixer_module.cpp#L777)), but in general the controller should always publish the setpoints, even when not armed (in this case uuv publishes only depending on the mode).

Reply:
Just for more context on this, relative to the work queue rates, here is what I see:

With (custom) `uuv_att_control` and `uuv_pos_control` running:
```
Work Queue: 11 threads                          RATE        INTERVAL
|__ 1) wq:rate_ctrl    
|   |__ 1) control_allocator                297.9 Hz         3357 us
|   |__ 2) dshot                            297.9 Hz         3357 us
|   |__ 3) uuv_att_control                  198.6 Hz         5035 us
```

With just `uuv_att_control` running:
```
Work Queue: 11 threads                          RATE        INTERVAL
|__ 1) wq:rate_ctrl    
|   |__ 1) control_allocator                199.0 Hz         5025 us
|   |__ 2) dshot                            199.0 Hz         5026 us
|   |__ 3) uuv_att_control                  198.9 Hz         5027 us
```

With just `uuv_pos_control` running:
```
Work Queue: 11 threads                          RATE        INTERVAL
|__ 1) wq:rate_ctrl    
|   |__ 1) control_allocator                 99.6 Hz        10042 us
|   |__ 2) dshot                             99.6 Hz        10043 us
```

With neither running:
```
Work Queue: 11 threads                          RATE        INTERVAL
|__ 1) wq:rate_ctrl    
|   |__ 1) control_allocator                  0.0 Hz            0 us
|   |__ 2) dshot                             20.5 Hz        48899 us
```

Reply:
How about conservative backup scheduling by default?

Reply:
Makes sense to me. 20Hz maybe?

Reply:
With backup scheduling in place:
```
|__ 1) wq:rate_ctrl    
|   |__ 1) control_allocator                 20.2 Hz        49475 us
|   |__ 2) dshot                             20.2 Hz        49477 us
|   \__ 3) vehicle_angular_velocity         400.5 Hz         2497 us
```
Let me know if I made the changes in the way you expected.